torch/_dynamo/trace_rules.py

```diff
@@ -35,40 +44,121 @@
     "torch._C.DisableTorchFunctionSubclass": TorchCtxManagerClassVariable,
     "torch.amp.autocast_mode.autocast": TorchCtxManagerClassVariable,
     "torch.autograd.grad_mode.enable_grad": TorchCtxManagerClassVariable,
     "torch.autograd.grad_mode.inference_mode": TorchCtxManagerClassVariable,
     "torch.autograd.grad_mode.no_grad": TorchCtxManagerClassVariable,
     "torch.autograd.grad_mode.set_grad_enabled": TorchCtxManagerClassVariable,
-    "torch.autograd.profiler.profile": TorchCtxManagerClassVariable,
-    "torch.autograd.profiler.record_function": TorchCtxManagerClassVariable,
     "torch.cpu.amp.autocast_mode.autocast": TorchCtxManagerClassVariable,
     "torch.cuda.amp.autocast_mode.autocast": TorchCtxManagerClassVariable,
-    "torch.profiler.profiler.profile": TorchCtxManagerClassVariable,
 }
 
 
+torch_name_rule_map = {**manual_torch_name_rule_map, **auto_torch_name_rule_map}
+
+
 @functools.lru_cache(None)
 def get_torch_obj_rule_map():
     d = dict()
     for k, v in torch_name_rule_map.items():
-        obj = load_object(k)
-        assert obj not in d
-        d[obj] = v
+        try:
+            obj = load_object(k)
+            d[obj] = v
+        except (AttributeError, ModuleNotFoundError):
+            pass
     return d
 
 
+def _load_obj_from_str(fully_qualified_name):
+    module, obj_name = fully_qualified_name.rsplit(".", maxsplit=1)
+    return getattr(importlib.import_module(module), obj_name)
+
+
 def load_object(name):
-    mod_name, obj_name = name.rsplit(".", 1)
-    mod = importlib.import_module(mod_name)
-    obj = getattr(mod, obj_name)
-    return obj
+    x = name.split("#")
+    if len(x) == 2:
+        obj = _load_obj_from_str(x[0])
+        val = getattr(obj, x[1])
+    else:
+        assert len(x) == 1
+        val = _load_obj_from_str(x[0])
+    if hasattr(val, "__wrapped__") and val is not torch.ops:
+        val = val.__wrapped__
+    return val
+
+
+@functools.lru_cache(None)
+def get_tensor_method():
+    s = set()
+    for name in dir(torch.Tensor):
+        method = getattr(torch.Tensor, name)
+        if isinstance(
+            method, (types.MethodDescriptorType, types.WrapperDescriptorType)
+        ):
+            s.add(method)
+    return s
+
+
+def is_in_graph_function(obj):
+    if obj in get_tensor_method() or isinstance(
+        obj,
+        (torch._ops.OpOverloadPacket, torch._ops.OpOverload),
+    ):
+        return True
+    if isinstance(
+        obj,
+        (
+            types.FunctionType,
+            types.MethodType,
+            types.BuiltinFunctionType,
+            types.MethodDescriptorType,
+            types.WrapperDescriptorType,
+        ),
+    ):
+        return is_allowed(obj)
+    else:
+        return False
 
 
 def lookup(obj):
     if not hashable(obj):
         return None
+    if id(obj) in _disallowed_function_ids:
+        return None
+    if hasattr(obj, "__wrapped__") and obj is not torch.ops:
+        obj = obj.__wrapped__
```

Comment:
If I wrapped a disallowed function id, it will become allowed. Is this intentional? Seems like a bug to me.

Reply:
This won't happen in the current code base, since we can't use ```allow_in_graph``` and ```disallow_in_graph``` as decorator inside of PT code base due to circular import. But I agree this potentially can lead to bug in the future, I have other ways to remove this unwrap, will update it.

Reply:
I leave as is now, since this caused CI flaky somehow, I'm still debugging it now.

Reply:
Moved ```is_user_defined_allowed``` and ```_disallowed_function_ids``` before the unwrap, so it's more safe now. But we should unify the user defined allowed with torch in graph functions in the future.